Formula/rust.rb

```diff
@@ -2,17 +2,16 @@ class Rust < Formula
   desc "Safe, concurrent, practical language"
   homepage "https://www.rust-lang.org/"
   license any_of: ["Apache-2.0", "MIT"]
-  revision 1
 
   stable do
-    url "https://static.rust-lang.org/dist/rustc-1.57.0-src.tar.gz"
-    sha256 "3546f9c3b91b1f8b8efd26c94d6b50312c08210397b4072ed2748e2bd4445c1a"
+    url "https://static.rust-lang.org/dist/rustc-1.58.1-src.tar.gz"
+    sha256 "a839afdd3625d6f3f3c4c10b79813675d1775c460d14be1feaf33a6c829c07c7"
 
     # From https://github.com/rust-lang/rust/tree/#{version}/src/tools
     resource "cargo" do
       url "https://github.com/rust-lang/cargo.git",
-          tag:      "0.58",
-          revision: "b2e52d7cab0a286ee9fcc0c17510b1e72fcb53eb"
+          tag:      "0.59.0",
+          revision: "7f08ace4f1305de7f3b1b0e2f765911957226bd4"
```

Comment:
technically the latest `cargo` is now one commit ahead of this, but it's untagged and only test changes: https://github.com/rust-lang/cargo/compare/0.59.0...rust-1.58.0